fedimint-build/src/lib.rs

```diff
@@ -1,21 +1,54 @@
 use std::env;
-use std::fs::File;
 use std::process::Command;
 
+/// Env variable to set to force git hash during build process
+const FORCE_GIT_HASH_ENV: &str = "FEDIMINT_BUILD_FORCE_GIT_HASH";
+
+/// Env variable the cargo will set during crate build to pass the detected git
+/// hash to the binary itself.
+const GIT_HASH_ENV: &str = "FEDIMINT_BUILD_CODE_VERSION";
+
+fn set_code_version_inner() -> Result<(), String> {
+    println!("cargo:rerun-if-env-changed={FORCE_GIT_HASH_ENV}");
+
+    if let Ok(hash) = env::var(FORCE_GIT_HASH_ENV) {
+        eprintln!("Forced hash via {FORCE_GIT_HASH_ENV} to {hash}");
+        println!("cargo:rustc-env={GIT_HASH_ENV}={hash}");
+        return Ok(());
+    }
+
+    let output = match Command::new("git").args(["rev-parse", "HEAD"]).output() {
```

Comment:
there is no `rerun-if` for this, means this will never get rerun

so 
```
cargo build
... commit stuff ...
cargo build
```

will not update the hash.

Reply:
@Maan2003 Oh. You're right. But is there anything we can do about it? Do we just need to force it to run on every build?

Reply:
rerunning on every build is sad, it will remove all intecremental compilation :/

maybe we can just afford this being incorrect?

Reply:
I'm not sure if we really care. E.g. right now we don't record anywhere a dirty hash, so just because the local binary has a given hash, doesn't mean it wasn't built with something far, far detached from it.

Nix-built binaries is what really matters.

Reply:
and in nix we are not even setting the hash :smile:

Reply:
@Maan2003 

```
    // Note: best effort approach to force a re-run when the git hash in
    // the local repo changes without wrecking the incremental compilation
    // completely.
    if Path::new(".git/HEAD").exists() {
        println!("cargo:rerun-if-changed=.git/HEAD");
    }
    // Common(?) `git workdir` setup
    if Path::new("../HEAD").exists() {
        println!("cargo:rerun-if-changed=../HEAD");
    }
```

LGTY?

Reply:
I don't know about workdir. but what is .git/HEAD relative to?

Reply:
Ah, damn it. Right. @Maan2003 How about now?